lib/net.js

```diff
@@ -872,6 +873,10 @@ function internalConnect(
 
   var err;
 
+  if (typeof address !== 'string') {
```

Comment:
Can we move this to `.connect()` instead, under the `if (pipe)` check? `internalConnect()` is also called for non-pipe connections and so it would be an unnecessary/redundant check in those cases.